Clients/Xamarin.Interactive.Client/Telemetry/Client.cs

```diff
@@ -6,92 +6,112 @@
-
-                        if (serverProc != null)
-                            // FIXME: server process to provide endpoint somehow
-                            telemetryEndpoint = "http://localhost:5000/api/";
-                    } catch {
-                    }
-                }
-
-                if (!Prefs.Telemetry.Enabled.GetValue () ||
-                    !Uri.TryCreate (telemetryEndpoint, UriKind.Absolute, out var telemetryApiUri)) {
+                if (!Prefs.Telemetry.Enabled.GetValue ()) {
                     Log.Info (TAG, "Telemetry is disabled");
                     return;
                 }
 
                 enabled = true;
 
-                httpClient = new HttpClient {
-                    BaseAddress = telemetryApiUri
+                // InMemoryChannel is the default channel, but we set it up manually here so we can tweak several
+                // default settings that are undesirable for desktop apps.
+                channel = new InMemoryChannel {
+                    // Defaults to 30s, but since we are changing buffer.Capacity to 1, we can make this infinite and
+                    // avoid pointlessly waking up InMemoryTransmitter's Runner.
+                    SendingInterval = Timeout.InfiniteTimeSpan,
                 };
 
-                httpClient.DefaultRequestHeaders.UserAgent.Add (
-                    new ProductInfoHeaderValue (
-                        "XamarinInteractiveTelemetry",
-                        BuildInfo.VersionString));
+                // There is no reasonable public API for changing the buffer capacity at this time.
+                // You can achieve it by turning on DeveloperMode, but that has other consequences.
+                // So we reflect.
+                //
+                // The default Capacity is 500, which is far too large for us (and perhaps most non-server apps).
+                // We want to avoid having to perform a blocking Flush call on the UI thread, and since our events
+                // are currently few and far between, we set Capacity to 1 to essentially get auto-flush.
+                var channelBuffer = typeof (InMemoryChannel)
+                    .GetField ("buffer", BindingFlags.NonPublic | BindingFlags.Instance)
+                    .GetValue (channel);
+                channelBuffer
+                    .GetType ()
+                    .GetProperty ("Capacity", BindingFlags.Public | BindingFlags.Instance)
+                    .SetValue (channelBuffer, 1);
+
+                var config = new TelemetryConfiguration ("@TELEMETRY_INSTRUMENTATION_KEY@", channel);
+
+                appInsightsClient = new TelemetryClient (config);
+
+                appInsightsClient.Context.Session.Id = sessionid.ToString ();
+                appInsightsClient.Context.Device.OperatingSystem = host.OSName.ToString ();
+
+                // TODO: Make these GlobalProperties when we bump to 2.7.0-beta3 or later
```

Comment:
Please store the dictionary as a local for all of these adds.

1. So you don't have to change 50 of them when you can use GlobalProperties
2. so each Add doesn't have to go through all the property dereferences each time

Reply:
Also please use the same formatting for all properties (each key/value pair on a separate line)

Reply:
Fixed.